src/app/services/status.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StatusService {
+
+  isLoggedIn = new BehaviorSubject<boolean>(false);
+ checkStatus = this.isLoggedIn.asObservable();
```

Comment:
For this use case, creating a simple fn to check the auth status should suffice,

`isLoggedin = () => localStorage.getItem('token') !== null`